### ISSUE TITLE
Forge Unintentionally Overrides Some Vanilla Block Replacement

### ISSUE THREAD

**Issue by JoelleEmmily** [MinecraftForge/MinecraftForge#8319]
**Minecraft Version:** 1.18.1

**Forge Version:** 39.0.8

**Steps to Reproduce:**
 1. Lay cluster of moss blocks with a single line of dirt blocks as indicator
 2. Place 2x2 spruce sapling cluster
 3. Maturate naturally or with bone meal

**Description of issue:**
Forge overrides podzol replacement when 2x2 spruce sapling cluster matures on a moss blocks. Other blocks, such as dirt, are replaced as normal. Issue occurs with or without modes installed, issue is not reproducible on clean Minecraft install. Blocks directly under tree is replaced, other block replacement issues have not been notice, but exhaustive testing was not carried out.

**Comment by JoelleEmmily** [comment]
![2021-12-22_22 16 13](https://user-images.githubusercontent.com/3618013/147184937-4587d5c4-5edf-42dc-87c9-062da49a8e08.png)
![2021-12-22_22 16 22](https://user-images.githubusercontent.com/3618013/147184939-332e740e-cb26-496b-aa8d-eee4b5962d6e.png)
![2021-12-22_22 19 00](https://user-images.githubusercontent.com/3618013/147184941-c7bd517e-8cea-4412-af2e-11c4adad22b9.png)
![2021-12-22_22 22 16](https://user-images.githubusercontent.com/3618013/147184944-2a66792b-4eba-4e07-91e7-c8a059bda935.png)
[debug.log](https://github.com/MinecraftForge/MinecraftForge/files/7766540/debug.log)
[latest.log](https://github.com/MinecraftForge/MinecraftForge/files/7766541/latest.log)

**Comment by alcatrazEscapee** [comment]
Podzol is placed by `AlterGroundDecorator`, which calls `Feature.isGrassOrDirt` which calls `Feature.isDirt` to check if a given block can be replaced.

Forge patches the following, changing the tag from referencing the vanilla tag `minecraft:dirt` to the tag `forge:dirt`
https://github.com/MinecraftForge/MinecraftForge/blob/0320c185b534a014d02573b5c021b0061fec0ec1/patches/minecraft/net/minecraft/world/level/levelgen/feature/Feature.java.patch#L22

Contents of vanilla `minecraft:dirt` block tag:
```json
{
  "replace": false,
  "values": [
    "minecraft:dirt",
    "minecraft:grass_block",
    "minecraft:podzol",
    "minecraft:coarse_dirt",
    "minecraft:mycelium",
    "minecraft:rooted_dirt",
    "minecraft:moss_block"
  ]
}
```

Contents of the Forge `forge:dirt` block tag:

```json
{
  "replace": false,
  "values": [
    "minecraft:dirt",
    "minecraft:grass_block",
    "minecraft:coarse_dirt",
    "minecraft:podzol",
    "minecraft:mycelium"
  ]
}
```

---

imo, the Forge tag should just be a reference to `#minecraft:dirt`.

**Comment by sciwhiz12** [comment]
Confirmed to exist on 1.18.1-39.0.8.

This seems like this could be resolved by #7891, but it seems the PR doesn't add the moss block in particular to the `forge:dirt` tag (only rooted dirt). I agree with @alcatrazEscapee's opinion that the forge tag should probably just be a reference to `minecraft:dirt` instead, with the possibility of phasing out the forge tag in favor of the minecraft tag (though that may be inadvisable because of the possibility it may break mods, but we are in the breaking changes window still). @SuperMartijn642, could you look into that before your PR gets merged?
